src/org/jrubyparser/ast/SafeOpAsgnNode.java

```diff
@@ -0,0 +1,134 @@
+/*
```

Comment:
This should add a boolean isLazy like we have in JRuby core.  all call/fcall/vcalls can all be lazy (e.g. lonely).    If we went the route of making safe nodes then we would need to double anything which can be on the right side of the lonely operator.